documentation/README.md

```diff
@@ -0,0 +1,80 @@
+
+Depending on the project, sometimes we might decide to have a single notebook be the "Getting started" tutorial; other times, we might have multiple ones, for example, one for each module.
+
+### Guidelines
+
+doc/requirements.txt (or doc/environment.yml) should contain all the dependencies required to run the notebooks with notes specifying if a specific library is only used for a particular notebook so we can remove it if we remove the notebook.
+
+### User Guide
+
+The User Guide aims to bring the user to a proficiency level where they feel comfortable with the library. After learning the User Guide, they should be able to perform tasks, tweak the simplest parameters, and understand the project's structure.
+
+As with "Getting Started", these are Jupyter notebooks, and we'll decide on a case-by-case basis for organizing them: if it'll be a single notebook or multiple ones.
+
+In cases where the library requires technical concepts (sklearn-evaluation is an excellent example since the docs assume it is familiar with Machine Learning concepts), the User Guide should explain these concepts. Explaining concepts in detail and with clarity will help us acquire users via search engines like Google.
+
+### Cookbook
+
+These are code snippets that users can copy-paste. We should pay attention to finding common patterns; once we identify them, we can create snippets and add them to the cookbook so users can quickly find them and get things done.
+
+A great idea to find patterns is to search on Stack Overflow for popular questions. Finding popular Stack Overflow questions will tell us what people are trying to do. For example, sklearn-evaluation provides a function to plot a confusion matrix; if we search "confusion matrix" we'll find questions such as "How do I plot a normalized confusion matrix?" or "How do I plot a multiclass confusion matrix?", both of them are good candidates to be included as recipes in our cookbook.
+
+If the recipes are simple, they should be included in the docstrings of the relevant functions/classes (and they [should be tested as part of the CI](https://docs.pytest.org/en/7.1.x/how-to/doctest.html)). When the recipes are more ellaborated (e.g., they require some setup that needs explanation, other packages, or multiple modules), we can create a *Cookbook* section in the documentation and add them as notebooks.
+
+### API Reference
+
+- All public functions and methods should be documented. We use the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) format
+- At least one example (that users can copy paste) should be provided
+
+## Tools
+
+We're migrating all our documentation websites to [Jupyterbook](https://jupyterbook.org/en/stable/intro.html); however, some still use Sphinx.
+
+## General Guidelines
+
+### Style
+
+Writing documentation is hard, so ensure you proofread several times and use tools like Grammarly to simplify writing.
```

Comment:
This might drop the moment we add spellcheck as part of the CI

Reply:
I think that should be the default tbh

Reply:
spellcheck != style. we can add spellcheck, but style is also important